### PR TITLE
Efficient Merkle Tree Formation Alg

### PR DESCRIPTION
**Pull request by yehjxraymond**

### CODE REVIEW

**Comment by yehjxraymond** [comment]
Bug submitted by Wu Yong:
Unable to batch for one certificate due to a logic error. Fix before merging.

File: src/batchVerify.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+const batchVerify = async undigestedCertDir => {
+  const certFileNames = await certificatesInDirectory(undigestedCertDir);
+  let allVerified = true;
+  certFileNames.forEach(file => {
+    const certificate = readCert(undigestedCertDir, file);
+    const verified = verifySignature(certificate);
+    allVerified = allVerified && verified;
+    if (verified) {
+      logger.debug(`${file}: Verified`);
```
Comment (rjchow) [inline src/batchVerify.js:13]: logger.info instead

File: src/crypto.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+const ethereumjsUtil = require("ethereumjs-util");
+
+/*
+* FUNCTIONS TAKEN FROM OPEN_ATTESTATION FRAMEWORK
```
Comment (rjchow) [inline src/crypto.js:4]: Add openattestation utils to its package.json's files and import them instead

https://stackoverflow.com/questions/33016714/how-to-require-a-file-other-than-the-npm-main-file-in-node

**Pull request merged**